packages/ra-ui-materialui/src/button/DeleteButton.stories.tsx

```diff
@@ -0,0 +1,45 @@
+import * as React from 'react';
+import { createTheme } from '@mui/material';
+import { DeleteButton } from './DeleteButton';
+import { AdminContext } from '../AdminContext';
+import frenchMessages from 'ra-language-french';
+import polyglotI18nProvider from 'ra-i18n-polyglot';
+import englishMessages from 'ra-language-english';
+
+const theme = createTheme({
+    palette: {
+        error: {
+            main: '#07BA8F',
+        },
+    },
+});
+
+const i18nProvider = polyglotI18nProvider(
+    locale => (locale === 'fr' ? frenchMessages : englishMessages),
+    'en'
+);
+
+export default { title: 'ra-ui-materialui/button/DeleteButton' };
+
+export const Basic = () => (
+    <AdminContext>
+        <DeleteButton label="Delete" record={{ id: 1 }} />
+    </AdminContext>
+);
+
+export const Pessimistic = () => (
+    <AdminContext i18nProvider={i18nProvider}>
+        <DeleteButton
+            mutationMode="pessimistic"
+            record={{ id: 1 }}
+            label="Delete"
+            resource="post"
+        />
+    </AdminContext>
+);
+
+export const WithUserDefinedPalette = () => (
```

Comment:
none of the stories actually test the case of the issue, along with the `color` props you changed. You should at least have one. E.g.:

```tsx
const theme2 = createTheme({
    palette: {
        primary: {
            light: colors.orange[100],
            main: colors.orange[500],
            contrastText: colors.grey[50],
        },
        error: {
            main: colors.orange[500],
        },
    },
});

export const ContainedWithUserDefinedPalette = () => (
    <AdminContext theme={theme2}>
        <DeleteButton
            variant="contained"
            color="primary"
            label="Delete"
            record={{ id: 1 }}
        />
    </AdminContext>
);
```

(with `colors` imported from MUI)